### ISSUE TITLE
Version in user agent is wrong when running from master

### ISSUE THREAD

**Issue by mikalstill** [shakenfist/shakenfist#286]
We use git release tags to calculate this version, and the most recent tag on that branch os v0.2.1.

**Comment by mandoonandy** [comment]
This appears to be a problem emanating from git. I can see the problem but cannot replicate it. ie. I cannot create a tag that fails to be seen by pbr like the current tags v0.2.1, v0.2.2, v0.2.3, v0.2.4.

I used `python3 setup.py --version` to check  the version seen by setup/pbr.

`pbr` only looks at annotated tags. The release instructions suggest to tag using `-s`. Using `-s` to sign a tag obviously annotates the tag. So the ignored tags are annotated as required by pbr...

**Comment by mikalstill** [comment]
I had assumed its because the signed tags are on the v0.2-devel branch, and the latest tag on master is v0.2.1 or something. So when I push master somewhere I get a really old tag. Your new branching strategy would largely resolve this I think?

**Comment by mandoonandy** [comment]
I was going to say "All the tags in are master. They can be seen with `git log --tags`"
### Right-O  (Stuff I learnt - Stuff you probably already know.)
`git log --tags` lists all the tags. Even if those tags are not on the branch.
Our problem occurs when a commit is only tagged on devel-v0.x branch. *As you said above.*
My confusion occurred because when I used `--tags` I saw the tag despite it not being on the branch AND my `log` alias hid the annotation that it was from a different branch.

### TL;DR
Tag in master. 
Confirmed that this will not happen in new branching strategy.
We need to tag a new release to fix this. Cannot use `v0.2.4` since it has been stolen by the dastardly devel-v0.2 branch.

**Comment by mikalstill** [comment]
I should upload my hotfixes and we can tag that as v0.2.next

**Comment by mandoonandy** [comment]
Excellent. I'll close this on the hope that your next release will solve it.
